### PR TITLE
CP-18025: Change blktap license from GPLv2 to BSD

### PR DESCRIPTION
**Pull request by chandrikas**
All files in this package (barring exceptions noted below) are now
licensed under a 3-clause BSD license. Please see the LICENSE file
for details.
    
The files licensed under other/different licenses are:
    
    blktap/
       |-- autogen.sh
       |-- driver/
       |      |-- atomicio.c
       |      |-- atomicio.h
       |      |-- md5.c
       |      |-- md5.h
       |
       |--include/
       |      |-- list.h
       |
       |--vhd/
           |--lib/
               |-- atomicio.c
               |-- atomicio.h
    
Please refer to the individual source file headers for how a particular
file is licensed.
    
Third party contributors [@antonym and @cristiklein] who contributed to
 blktap under the GPLv2 have been consulted and their consent has been
 obtained.
    
Signed-off-by: Chandrika Srinivasan \<chandrika.srinivasan@citrix.com\>
Signed-off-by: Cristian Klein \<cristian.klein@cs.umu.se\>
Signed-off-by: Antony Messerli \<antony@mes.ser.li\>
Acked-by: Germano Percossi \<germano.percossi@citrix.com\>
Acked-by: Lars Kurth \<lars.kurth@citrix.com\>
[On behalf of Citrix employees]

### CODE REVIEW

**Comment by germanop** [comment]
* We need publicly written consent by the external contributors or we'll remove their patches before proceeding
* I would keep the LICENSE file in place with the BSD license, stating that the whole project is meant to be BSD 3-clause. This is useful for 2 reasons:
** If in the future some files will have an exceptional license, this exception can be stated (along with the file itself) in the LICENSE file
** If we should find out in the future we missed some files or there are inconsistencies, we do not need to amend them contacting the developers that already acked this commit, because the purpose of the commit is stated clearly in the LICENSE file
* The commit message should be a bit more verbose about the purpose (we are changing the whole
project to be 3-clause BSD) to avoid future problems as per the above point
* I am waiting for explicit consent to ack on behalf of Citrix employees

**Comment by germanop** [comment]
This is the list of committers after the license change
```
Andrei Lifchits <andrei.lifchits@citrix.com>
Andrew Cooper <andrew.cooper3@citrix.com>
Antony Messerli <amesserl@rackspace.com>
Bob Ball <bob.ball@citrix.com>
Chandrika Srinivasan <chandrika.srinivasan@citrix.com>
Cristian KLEIN <cristiklein@gmail.com>
David Vrabel <david.vrabel@citrix.com>
Felipe Franciosi <felipe@paradoxo.org>
Flavien Quesnel <flavien.quesnel@citrix.com>
Frediano Ziglio <frediano.ziglio@citrix.com>
Germano Percossi <germano.percossi@citrix.com>
John Else <john.else@citrix.com>
Jonathan Davies <jonathan.davies@citrix.com>
Jon Ludlam <jonathan.ludlam@citrix.com>
Jon Ludlam <jonathan.ludlam@eu.citrix.com>
Jorge Martin <jorge.martin@citrix.com>
Kaifeng Zhu <kaifeng.zhu@citrix.com>
Kostas Ladopoulos <konstantinos.ladopoulos@citrix.com>
Mate Lakat <mate.lakat@citrix.com>
philippe gabriel <philippe.gabriel@citrix.com>
Pritha Srivastava <pritha.srivastava@citrix.com>
Rafal Mielniczuk <rafal.mielniczuk@citrix.com>
Ross Lagerwall <ross.lagerwall@citrix.com>
Siddharth Vinothkumar <siddharth.vinothkumar@citrix.com>
Stefano Panella <Stefano.Panella@citrix.com>
Thanos Makatos <thanos.makatos@citrix.com>
Thanos <thanos.makatos@citrix.com>
vineetht <vineeth.thampi@citrix.com>
Zheng Li <dev@zheng.li>
Zheng Li <zheng.li@eu.citrix.com>
```

Only Antony Messerli <amesserl@rackspace.com> and Cristian KLEIN <cristiklein@gmail.com>
were not Citrix employees when code was submitted and we need to have explicit consent

**Comment by germanop** [comment]
In the LICENSE file we need to make clear that anything apart from atomic.c, atomic.h, autogen.sh (BSD 2-clause) are BSD 3-clause.

File: drivers/tapdisk-nbdserver.h [hunk 1]
```diff
@@ -38,10 +38,39 @@ typedef struct td_nbdserver_client td_nbdserver_client_t;
...
+	char handle[8];
+	uint64_t from;
+	uint32_t len;
+} __attribute__ ((packed));
+
+struct nbd_reply {
+	uint32_t magic;
+	uint32_t error;		
+	char handle[8];		
+};
```
Comment (robertbreker) [inline drivers/tapdisk-nbdserver.h]: Shouldn't this be __attribute__ ((packed));?

Comment (chandrikas) [inline drivers/tapdisk-nbdserver.h]: The structure is already aligned (4 + 4 + 8) along a 4-byte boundary, so probably not required?

**Comment by germanop** [comment]
I think the License field in here mk/blktap.spec.in needs to change, as well.

**Comment by germanop** [comment]
Acked-by: Germano Percossi \<germano.percossi@citrix.com\>

**Comment by cristiklein** [comment]
Acked-by: Cristian Klein <cristiklein@gmail.com>

**Comment by antonym** [comment]
Acked-by: Antony Messerli <antony@mes.ser.li>

**Pull request closed without merging**